pyairtable/utils.py

```diff
@@ -21,7 +23,9 @@ def datetime_from_iso_str(value: str) -> datetime:
     Args:
         value: datetime string e.g. "2014-09-05T07:00:00.000Z"
     """
-    return datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%fZ")
+    if "Z" in value:
+        return datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%fZ")
+    return datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%f%z")
```

Comment:
For this one the change is also quite simple. The `%z` will already recognize the `Z` from the string. No need to differentiate.
```suggestion
    return datetime.strptime(value, "%Y-%m-%dT%H:%M:%S.%f%z")
```

Reply:
We could argue if we want the time by airtable to default to our local timezone, as it is by default send as UTC. But with the coming update that might not be the case anymore, so I would not touch that. Since all relevant information is available, and you can always call astimezone, this should be fine now

Reply:
Sounds good. Thank you!

Reply:
by the way: why dont we use this?
https://docs.python.org/3/library/datetime.html#datetime.time.fromisoformat

Reply:
That would make sense to me, though it also raises a question for me about how this library parses date fields (i.e., dates without times). Seems like it might not perform as desired